### PR TITLE
Avoid logging expected tracebacks while testing calls to the engine server api

### PR DESCRIPTION
**Pull request by ptormene**
This looks like a good compromise

### CODE REVIEW

File: openquake/server/tests/test_public_mode.py [hunk 1]
```diff
@@ -230,8 +231,10 @@ def test_abort(self):
 
     def test_err_1(self):
         # the rupture XML file has a syntax error
```
Comment (micheles) [inline openquake/server/tests/test_public_mode.py:233]: Please add a comment about why we are disabling the logs here, for future reference

File: .github/workflows/macos_intel_install.yml [hunk 2]
```diff
@@ -92,19 +92,19 @@ jobs:
         run: |
           source ~/openquake/bin/activate
           cd ~/work/oq-engine/oq-engine
```
Comment (micheles) [inline .github/workflows/macos_intel_install.yml:94]: Please add a comment about why we are using `-v 2` (I guess it is the verbosity level).

File: openquake/server/tests/test_public_mode.py (hunk 1 above)
Comment (ptormene) [reply openquake/server/tests/test_public_mode.py]: Done

File: .github/workflows/macos_intel_install.yml (hunk 2 above)
Comment (ptormene) [reply .github/workflows/macos_intel_install.yml:94]: Done

**Review by micheles — APPROVED** [review]

**Pull request merged**